### PR TITLE
netty, alts: expose ProtectedPromise, and writeBufferedAndRemove methods

### PR DESCRIPTION
**Pull request by creamsoup**
ProtectedPromise will be used in frameProtector in PSP.
PSP also needs to call writeBufferedAndRemove.

### CODE REVIEW

File: alts/src/main/java/io/grpc/alts/internal/ProtectedPromise.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 import static com.google.common.base.Preconditions.checkState;
 
+import io.grpc.Internal;
```
Comment (carl-mastrangelo) [inline alts/src/main/java/io/grpc/alts/internal/ProtectedPromise.java]: This is not sufficient.  You need to expose this via a class named Internal...

Comment (creamsoup) [reply alts/src/main/java/io/grpc/alts/internal/ProtectedPromise.java]: done. PTAL

File: netty/src/main/java/io/grpc/netty/WriteBufferingAndExceptionHandlerUtils.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+package io.grpc.netty;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelHandlerContext;
+
+/** Utility class for {@link WriteBufferingAndExceptionHandler}. */
+public final class WriteBufferingAndExceptionHandlerUtils {
```
Comment (carl-mastrangelo) [inline netty/src/main/java/io/grpc/netty/WriteBufferingAndExceptionHandlerUtils.java]: this needs to be hidden too

Comment (creamsoup) [reply netty/src/main/java/io/grpc/netty/WriteBufferingAndExceptionHandlerUtils.java]: done.

File: netty/src/main/java/io/grpc/netty/NettyClientHandler.java [hunk 3]
```diff
@@ -445,12 +445,7 @@ public void handleProtocolNegotiationCompleted(
     this.attributes = this.attributes.toBuilder().setAll(attributes).build();
     this.securityInfo = securityInfo;
     super.handleProtocolNegotiationCompleted(attributes, securityInfo);
-    // Once protocol negotiator is complete, release all writes and remove the buffer.
-    ChannelHandlerContext handlerCtx =
-        ctx().pipeline().context(WriteBufferingAndExceptionHandler.class);
-    if (handlerCtx != null) {
-      ((WriteBufferingAndExceptionHandler) handlerCtx.handler()).writeBufferedAndRemove(handlerCtx);
-    }
+    InternalWriteBufferingAndExceptionHandlerUtils.writeBufferingAndRemove(ctx().channel());
```
Comment (carl-mastrangelo) [inline netty/src/main/java/io/grpc/netty/NettyClientHandler.java]: nit: This cannot call into an internal package.   Youll need to expose writeBufferingAndRemove as a package-private static on the NCH itself, and make this call into NCH

Comment (creamsoup) [reply netty/src/main/java/io/grpc/netty/NettyClientHandler.java]: done.

**Pull request merged**